### PR TITLE
Adds New PushNotifications Stat

### LINKED ISSUE #3384: Notifications: Add Mixpanel Events
Whenever the app is launched form a Push Notification, we should hit a Mixpanel Event indicating the type of notification involved.

### PR DESCRIPTION
**Pull request by jleandroperez**
Closes #3384

/cc @aerych (Thanks in advance Eric!)

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.m [hunk 1]
```diff
@@ -317,6 +317,9 @@ - (void)showDetailsForNoteWithID:(NSString *)notificationID
     if (notification) {
         DDLogInfo(@"Pushing Notification Details for: [%@]", notificationID);
         
+        NSDictionary *properties = @{ @"type" : (notification.type ?: [NSString string]) };
+        [WPAnalytics track:WPAnalyticsStatPushNotificationAlertPressed withProperties:properties];
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.m]: Is it weird to pass a dictionary with an empty string for a type? Why not just pass nil for properties?

Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.m]: Type is expected to be always there (this is just a failsafe). But perhaps it makes more sense to move the check outside. Updating, thanks!

**Comment by aerych** [comment]
:shipit:  :)

**Comment by jleandroperez** [comment]
Thank you Eric!

File: WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerMixpanel.m [hunk 2]
```diff
@@ -379,6 +379,9 @@ - (WPAnalyticsTrackerMixpanelInstructionsForStat *)instructionsForStat:(WPAnalyt
             [instructions setSuperPropertyAndPeoplePropertyToIncrement:@"number_of_times_editor_published_post"];
             [instructions setCurrentDateForPeopleProperty:@"last_time_published_post"];
             break;
+        case WPAnalyticsStatPushNotificationAlertPressed:
+            instructions = [WPAnalyticsTrackerMixpanelInstructionsForStat mixpanelInstructionsForEventName:@"PushNotification - Alert Tapped"];
```
Comment (sendhil) [inline WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerMixpanel.m]: @jleandroperez could we adjust this to 'Push Notification'?

Comment (jleandroperez) [inline WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerMixpanel.m]: @sendhil sure! Updated [in this PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/3430/files)

**Pull request merged**